src/main/java/chopchop/model/ingredient/Ingredient.java

```diff
@@ -54,15 +104,37 @@ public Quantity getQuantity() {
      */
     public Ingredient combine(Ingredient other) throws IncompatibleIngredientsException {
 
-        if (!this.name.equals(other.name)) {
+        if (!this.isSame(other)) {
             throw new IncompatibleIngredientsException(String.format("cannot combine '%s' with '%s'",
                 this.name, other.name));
         }
 
-        // TODO: expiry date handling! see #58
-        return this.quantity.add(other.quantity)
-            .map(newQty -> new Ingredient(this.name.toString(), newQty, this.expiryDate))
-            .orElseThrow(IncompatibleIngredientsException::new);
+        // there's no constructor that takes both an existing map and the comparator...
+        var newSets = new TreeMap<Optional<ExpiryDate>, Quantity>(SET_COMPARATOR);
+        newSets.putAll(this.sets);
+
+        // because of exceptions, we cannot do this using nice lambdas and stuff.
+        // so write some dirty imperative code to merge the ingredients.
+        for (var entry : other.sets.entrySet()) {
```

Comment:
I'm just confused. What's the advantage of using `var` in general? Like in this case, wouldn't specifying the type make it clearer?

Reply:
less typing. as you know typing places strain on your wrist, which is bad for health <-> `var` is good for health

ok but most of the time, `var` is used either (a) when the type is obvious, like this:
```java
var foo = new SomeOverlyLongClassName();
// versus:
SomeOverlyLongClassName foo = new SomeOverlyLongClassName();
```

and (b) when the type is not important, like in this case. explicitly writing `Map.Entry<Optional<ExpiryDate>, Quantity>` is tedious and doesn't contribute anything to the clarity of the code.

Reply:
Oo Thanks. This is more helpful than my doctor who only told me not to use it if it hurts but never say how.